File: core/src/com/bombbird/terminalcontrol/ui/Shoreline.java
package com.bombbird.terminalcontrol.ui;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.utils.Array;
import com.bombbird.terminalcontrol.TerminalControl;
import com.bombbird.terminalcontrol.utilities.math.MathTools;
import com.bombbird.terminalcontrol.utilities.saving.FileLoader;

public class Shoreline {
    private static Array<Array<Integer>> landmasses = new Array<Array<Integer>>();

    public static void loadShoreline() {
        landmasses = FileLoader.loadShoreline();
    }

    public static void renderShape() {
        for (int i = 0; i < landmasses.size; i++) {
            for (int j = 2; j < landmasses.get(i).size; j += 2) {
                int prevX = landmasses.get(i).get(j - 2);
                int prevY = landmasses.get(i).get(j - 1);
                int thisX = landmasses.get(i).get(j);
                int thisY = landmasses.get(i).get(j + 1);

                if ((!MathTools.withinRange(prevX, 1260, 4500) || !MathTools.withinRange(prevY, 0, 3240)) && (!MathTools.withinRange(thisX, 1260, 4500) || !MathTools.withinRange(thisY, 0, 3240))) {
                    //Both points not inside range, don't draw line
                    continue;
                }
                //Draw lines to connect points
                TerminalControl.radarScreen.shapeRenderer.setColor(Color.BROWN);
                TerminalControl.radarScreen.shapeRenderer.line(prevX, prevY, thisX, thisY);
            }
        }
    }
}

File: core/src/com/bombbird/terminalcontrol/ui/Shoreline.kt
package com.bombbird.terminalcontrol.ui

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.utils.Array
import com.bombbird.terminalcontrol.TerminalControl
import com.bombbird.terminalcontrol.utilities.math.MathTools.withinRange
import com.bombbird.terminalcontrol.utilities.saving.FileLoader

object Shoreline {
    private val radarScreen = TerminalControl.radarScreen!!
    private var landmasses = Array<Array<Int>>()

    fun loadShoreline() {
        landmasses = FileLoader.loadShoreline()
    }

    fun renderShape() {
        for (i in 0 until landmasses.size) {
            var j = 2
            while (j < landmasses[i].size) {
                val prevX = landmasses[i][j - 2]
                val prevY = landmasses[i][j - 1]
                val thisX = landmasses[i][j]
                val thisY = landmasses[i][j + 1]
                if ((!withinRange(prevX, 1260, 4500) || !withinRange(prevY, 0, 3240)) && (!withinRange(thisX, 1260, 4500) || !withinRange(thisY, 0, 3240))) {
                    //Both points not inside range, don't draw line
                    j += 2
                    continue
                }
                //Draw lines to connect points
                radarScreen.shapeRenderer.color = Color.BROWN
                radarScreen.shapeRenderer.line(prevX.toFloat(), prevY.toFloat(), thisX.toFloat(), thisY.toFloat())
                j += 2
            }
        }
    }
}